### PR TITLE
fmt: fix pl trim methods and add tests

### PR DESCRIPTION
**Pull request by maximilianfridrich**

### CODE REVIEW

File: test/fmt.c [hunk 1]
```diff
@@ -1185,3 +1185,68 @@ int test_text2pcap(void)
 	mem_deref(mb);
 	return err;
 }
+
+
+int test_trim(void)
```
Comment (alfredh) [inline test/fmt.c:1190]: please add name of the module to the testcase, i.e. `test_fmt_trim()`

File: src/fmt/pl.c [hunk 2]
```diff
@@ -784,16 +784,20 @@ const char *pl_strstr(const struct pl *pl, const char *str)
...
-	while (!re_regex(pl->p, 1, "[ \t\r\n]")) {
-		++pl->p;
-		--pl->l;
-		if (!pl->l)
-			return EINVAL;
+	if (!pl_isset(pl))
+		return 0;
+
+	size_t i = 0;
+	while (i <= pl->l && isspace((unsigned char)pl->p[i])) {
```
[the remark this answers is not in the document]
Comment (maximilianfridrich) [reply src/fmt/pl.c]: Good catch, thank you. Done.

File: test/fmt.c (hunk 1 above)
Comment (maximilianfridrich) [reply test/fmt.c]: Done.

File: src/fmt/pl.c [hunk 3]
```diff
@@ -784,16 +784,20 @@ const char *pl_strstr(const struct pl *pl, const char *str)
...
+	if (!pl_isset(pl))
+		return 0;
+
+	size_t i = 0;
+	while (i < pl->l && isspace((unsigned char)pl->p[i])) {
+		++i;
 	}
 
+	pl->p += i;
+	pl->l -= i;
```
Comment (alfredh) [inline src/fmt/pl.c:799]: optional: `pl_advance` can be used instead of these two lines ...

Comment (maximilianfridrich) [reply src/fmt/pl.c]: Perfect, thank you for the suggestion. Done.

**Review by cspiel1 — APPROVED** [review]

**Pull request merged**